libs/dst/ProcPathStasCollection.py

```diff
@@ -0,0 +1,154 @@
+        LOGGER.info("checking for installation required modules")
+        for worker_node in self.worker_node_list:
+            resp = worker_node.execute_cmd("pip list")
+            LOGGER.debug(resp)
+            for module in required_modules:
+                if module in str(resp):
+                    LOGGER.info("already installed module {}".format(module))
+                else:
+                    retry = 0
+                    while retry < 2:
+                        try:
+                            LOGGER.info("installing {}".format(module))
+                            worker_node.execute_cmd("pip install {}".format(module))
+                            break
+                        except IOError as err:
+                            LOGGER.info("facing error {} while installing {}".format(err, module))
+                            LOGGER.info("retrying installation of {}".format(module))
+                            retry += 1
+                    if retry >= 2:
+                        assert False
+
+    def start_collection(self):
+        """trigger command for collection stats"""
+        LOGGER.debug("starting stat collection")
+        for worker in self.worker_node_list:
+            file_name = "{}_{}.sqlite".format(str(worker.hostname).replace('.colo.seagate.com', ''),
+                                              str(datetime.now().strftime("%m_%d_%Y_%H_%M_%S")))
+            self.worker_stat_files_dict[file_name] = worker
+            self.stat_collection.append(Process(target=self.stat_collection_cmd,
+                                                args=(worker, proc_cmd.format(file_name))))
+        self.collect_pids(self.worker_stat_files_dict)
+        for proc in self.stat_collection:
+            proc.start()
+
+    def stop_collection(self):
+        """stop collection"""
+        for proc in self.stat_collection:
+            if proc.is_alive():
+                proc.kill()
+        LOGGER.debug("stopping stat collection")
+
+    def validate_collection(self):
+        """validate collection
+        check for sqlite files generated
+        """
+        LOGGER.debug("checking if stat collection is alive")
+        for proc in self.stat_collection:
+            if not proc.is_alive():
+                return False, "process is not alive"
+        for file_name, worker in self.worker_stat_files_dict.items():
+            if not worker.path_exists(file_name):
+                return False, "log files are missing"
+        return True, "process and logs are being generated"
+
+    def get_stat_files_to_local(self):
+        """function to collect generated logs and copy them back to local"""
+        for file_name, worker in self.worker_stat_files_dict.items():
+            if not worker.path_exists(file_name):
+                return False, "log files are missing"
+            else:
+                resp = worker.copy_file_to_local(remote_path=file_name, local_path=file_name)
```

Comment:
do we want to create one log dir and copy files there instaed?

Reply:
moved all logs to one single location.